### ISSUE TITLE
Missing attribute for device posture rules

### ISSUE THREAD

**Issue by bmoller** [cloudflare/cloudflare-go#670]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the library and it is still present.

### cloudflare-go version

0.19.0

### Go environment

GOVERSION="go1.15"

### Expected output

A usable device posture rule should be created for reference by Cloudflare for Teams policies.

### Actual output

The Go library currently does not pass a `flavor` as part of the rule `input`; this behavior differs from the Cloudflare for Teams web console. A usable JSON object should look like this:

```
{
    "type": "application",
    "name": "Jamf",
    "schedule": "5m",
    "match": [
        {
            "platform": "mac"
        }
    ],
    "input": {
        "path": "/Library/Application Support/JAMF/Jamf.app/Contents/MacOS/JamfDaemon.app/Contents/MacOS/JamfDaemon",
        "running": true,
        "thumbprint": "e86867eab7456a4fefcda5541be7d7e2c5aacbe9",
        "flavor": "application"
    }
}
```

The `flavor` is not currently defined in the rule struct: https://github.com/cloudflare/cloudflare-go/blob/master/device_posture_rule.go#L28:L36

It is admittedly also missing from the Cloudflare API docs, which is unfortunate.

I am not a direct consumer of the Cloudflare Go library but it's used by Terraform, which I use to manage our Cloudflare account. It doesn't make much sense to open a bug for the Cloudflare provider until it's fixed here because they use this library for API calls: https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/cloudflare/resource_cloudflare_device_posture_rule.go

### Code demonstrating the issue

{
    "type": "application",
    "name": "Jamf",
    "schedule": "5m",
    "match": [
        {
            "platform": "mac"
        }
    ],
    "input": {
        "path": "/Library/Application Support/JAMF/Jamf.app/Contents/MacOS/JamfDaemon.app/Contents/MacOS/JamfDaemon",
        "running": true,
        "thumbprint": "e86867eab7456a4fefcda5541be7d7e2c5aacbe9"
    }
}

### Steps to reproduce

1. `POST` the above JSON object to `client/v4/accounts/.../devices/posture` to create a new device posture rule
2. Note that the rule is displayed in the Cloudflare for Teams console list of rules, but does _not_ appear as an option when creating policies
3. `POST` the above JSON object again with `"flavor": "application"` in the `input` attribute to `client/v4/accounts/.../devices/posture/...`
4. Note that the rule is both listed in the console and may now be selected as an option for policies

**Comment by jacobbednarz** [comment]
by the looks, this field doesn't have public API support yet so once it's documented (and confirmed to be intentionally used by the public), we can look at adding it

**Comment by jacobbednarz** [comment]
Closed by #676, #677
